pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -454,6 +477,40 @@ func (r *ReconcileLogStorage) Reconcile(ctx context.Context, request reconcile.R
 			return reconcile.Result{}, err
 		}
 
+		// Check that if the ES Gateway certpair secret exists that it is valid (has key and cert fields)
+		// If it does not exist then this function returns a nil secret but no error and a self-signed
+		// certificate will be generated when rendering below.
+		tlsSecret, err = utils.ValidateCertPair(r.client,
+			rmeta.OperatorNamespace(),
+			esgateway.EsGatewayTLSSecret,
+			esgateway.EsGatewaySecretKeyName,
+			esgateway.EsGatewaySecretCertName,
+		)
+
+		// An error is returned in case the read cannot be performed of the secret does not match the expected format
+		// In case the secret is not found, the error and the secret will be nil.
+		if err != nil {
+			r.status.SetDegraded("Error validating ES Gateway TLS certificate", err.Error())
+			return reconcile.Result{}, err
+		}
+
+		if tlsSecret == nil {
```

Comment:
This is not certificate management compatible. You may want to throw an error if es-gateway and certificate management are used at the same time.

Reply:
Does this code mean that every cluster will have an es-gateway unless it is a managed cluster?

Reply:
> Does this code mean that every cluster will have an es-gateway unless it is a managed cluster?

Yes exactly.

> This is not certificate management compatible. You may want to throw an error if es-gateway and certificate management are used at the same time.

I plan to revisit this now that we've decided to re-purpose `tigera-secure-elasticsearch-cert` to be the gateway secret and I'll make sure that this is cert management compatible.